x/ecocredit/client/tx.go

```diff
@@ -49,17 +56,152 @@ func txCreateClass() *cobra.Command {
 			}
 			b, err := base64.StdEncoding.DecodeString(args[2])
 			if err != nil {
-				return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "base64_metadata is malformed, proper base64 string is required")
+				return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "metadata is malformed, proper base64 string is required")
 			}
 
-			c, ctx, err := mkMsgClient(cmd)
+			c, err := newMsgSrvClient(cmd)
 			if err != nil {
 				return err
 			}
-			_, err = c.CreateClass(cmd.Context(), &ecocredit.MsgCreateClassRequest{
+			msg := ecocredit.MsgCreateClassRequest{
 				Designer: args[0], Issuers: issuers, Metadata: b,
-			})
-			return mkTx(ctx, cmd, err)
+			}
+			_, err = c.client.CreateClass(cmd.Context(), &msg)
+			return c.send(err)
+		},
+	}
+	flags.AddTxFlagsToCmd(cmd)
+	return cmd
+}
+
+func txCreateBatch() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create_batch [issuer] [class_id] [metadata] [issuance]",
+		Short: "Issues a new credit batch",
+		Long: `Issues a new credit batch.
+
+Parameters:
+  issuer:    issuer address
+  class_id:  credit class
+  metadata:  base64 encoded issuance metadata
+  issuance:  JSON encode issuance list,
```

Comment:
this is a little awkward in the CLI no? hopefully we get some actual user interfaces for this soon

Reply:
Why it's awkward? IMHO using a readable and widely used text format is the best choice for a list of complex object.

(btw, I've changed it to YAML)

Reply:
I think what @aaronc means is that stepping back, this kind of functionality is quite cumbersome to do via CLI and we long term should be prioritizing non-cli based tx interfaces for functionality like this with so many parameters.